packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -60,21 +57,21 @@ class ShellRouteConfig extends RouteBaseConfig {
 
   @override
   Iterable<String> classDeclarations() {
-    if (routeDataClass.unnamedConstructor == null) {
+    if (routeDataClass.unnamedConstructor2 == null) {
       throw InvalidGenerationSourceError(
         'The ShellRouteData "$_className" class must have an unnamed constructor.',
         element: routeDataClass,
       );
     }
 
-    final bool isConst = routeDataClass.unnamedConstructor!.isConst;
+    final bool isConst = routeDataClass.unnamedConstructor2!.isConst;
```

Comment:
same https://pub.dev/documentation/analyzer/latest/dart_element_element/InterfaceElement/unnamedConstructor2.html

Reply:
and all other places